native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ListenerUtils.java

```diff
@@ -167,16 +169,38 @@ public static Object detach(Environment environment, BObject listenerBObject, BO
     private static void declareQueueIfNotExists(BObject service, Channel channel) throws IOException {
         BMap serviceConfig = (BMap) ((AnnotatableType) TypeUtils.getType(service))
                 .getAnnotation(StringUtils.fromString(ModuleUtils.getModule().getOrg() + ORG_NAME_SEPARATOR
-                                                              + ModuleUtils.getModule().getName() + VERSION_SEPARATOR
-                                                              + ModuleUtils.getModule().getVersion() + ":"
-                                                              + RabbitMQConstants.SERVICE_CONFIG));
+                        + ModuleUtils.getModule().getName() + VERSION_SEPARATOR
+                        + ModuleUtils.getModule().getVersion() + ":"
+                        + RabbitMQConstants.SERVICE_CONFIG));
         String queueName;
         if (service.getNativeData(RabbitMQConstants.QUEUE_NAME.getValue()) != null) {
+            // Queue name is given as the service name
             queueName = (String) service.getNativeData(RabbitMQConstants.QUEUE_NAME.getValue());
         } else {
+            // Queue name and config given in the service config
             queueName = serviceConfig.getStringValue(RabbitMQConstants.QUEUE_NAME).getValue();
+            Map<String, Object> argumentsMap = new HashMap<>();
+            boolean durable = false;
```

Comment:
What happens if the user has defined a service-name and `queueName` in the `rabbitmq:ServiceConfig` annotation ?

Reply:
Check this one WDYT db6ca785cc56b48625ed3d518c5b27c94ee088d0

Reply:
basically config replaced service name